Need a way to enable NODE_ENV="development for debugging.

I'm facing some problem about prerendering:

```
Uncaught exception: workerd/jsg/_virtual_includes/iterator/workerd/jsg/value.h:1480: failed: remote.jsg.Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.
```

I want to find out what's wrong, but there's no way in waku to turn on `NODE_ENV="development"` when `waku build`. `NODE_ENV="development" waku build` results in `Warning: NODE_ENV is set to 'development', but overriding it to 'production'.`, and in `waku.config.ts`, `mode="development"` is no op. Waku has no such cli param.

Do you think this is a valid feature request? Can you offer me a way to debug? Thank you a lot.

> So, do you want to use the dev build for the production? We may consider not overriding for such a special case. [@hi-ogawa](https://github.com/hi-ogawa) What do you think?

This is the direct way to debug. If there's an alternative way to debug, it's okay too.

In the future, vite will support bundle only mode, which may help debug where dev works but prod fails, but now waku fails when `"bundledDev: true"`.

```
✘ Build error: Build failed with 1 error:

[plugin waku:vite-plugins:html-shell]
AssertionError: The expression evaluated to a falsy value:

  assert(this.environment.mode === 'build')
AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert(this.environment.mode === 'build')

    at PluginContextImpl.resolveId (file:///home/hash/workdir/hashland/node_modules/.pnpm/waku@1.0.0-beta.2_@rolldown+plugin-babel@0.2.3_@babel+core@7.29.7_rolldown@1.0.3_vite@8_d188a23c66efea10cdbb2074af38ec07/node_modules/waku/dist/lib/vite-plugins/html-shell.js:31:17)
    at PluginContextImpl.handler (file:///home/hash/workdir/hashland/node_modules/.pnpm/vite@8.0.16_@types+node@25.9.1_esbuild@0.27.3_jiti@2.7.0_yaml@2.9.0/node_modules/vite/dist/node/chunks/node.js:33478:13)
    at plugin (file:///home/hash/workdir/hashland/node_modules/.pnpm/rolldown@1.0.3/node_modules/rolldown/dist/shared/bindingify-input-options-ClrST5Xx.mjs:1092:30)
    at plugin.<computed> (file:///home/hash/workdir/hashland/node_modules/.pnpm/rolldown@1.0.3/node_modules/rolldown/dist/shared/bindingify-input-options-ClrST5Xx.mjs:1625:18)
```

It fails intentionally.

--- a/packages/waku/src/lib/vite-rsc/cmd-build.ts
+++ b/packages/waku/src/lib/vite-rsc/cmd-build.ts
@@ -1,13 +1,14 @@
 import * as vite from 'vite';
 import type { Config } from '../../config.js';
 import { combinedPlugins } from '../vite-plugins/combined-plugins.js';
-import { loadConfig, loadDotEnv, overrideNodeEnv } from './loader.js';
+import { loadConfig, loadDotEnv } from './loader.js';
 import type { PreviewServer } from './preview.js';
 
 loadDotEnv();
 
 export async function runBuild() {
-  overrideNodeEnv('production');
+  // before vite.runnerImport: https://github.com/vitejs/vite/issues/20299
+  process.env.NODE_ENV ??= 'production';
   const config = await loadConfig();
   const builder = await vite.createBuilder({
     configFile: false,

--- a/packages/waku/src/lib/vite-rsc/cmd-build.ts
+++ b/packages/waku/src/lib/vite-rsc/cmd-build.ts
@@ -7,7 +7,7 @@ import type { PreviewServer } from './preview.js';
 loadDotEnv();
 
 export async function runBuild() {
-  // before vite.runnerImport: https://github.com/vitejs/vite/issues/20299
+  // set NODE_ENV before vite.runnerImport: https://github.com/vitejs/vite/issues/20299
   process.env.NODE_ENV ??= 'production';
   const config = await loadConfig();
   const builder = await vite.createBuilder({

--- a/packages/waku/src/lib/vite-rsc/cmd-dev.ts
+++ b/packages/waku/src/lib/vite-rsc/cmd-dev.ts
@@ -2,7 +2,7 @@ import path from 'node:path';
 import * as vite from 'vite';
 import type { Config } from '../../config.js';
 import { combinedPlugins } from '../vite-plugins/combined-plugins.js';
-import { loadConfig, loadDotEnv, overrideNodeEnv } from './loader.js';
+import { loadConfig, loadDotEnv } from './loader.js';
 
 loadDotEnv();
 
@@ -79,7 +79,8 @@ async function startDevServer(
 }
 
 export async function runDev(flags: { host?: string; port?: string }) {
-  overrideNodeEnv('development');
+  // before vite.runnerImport: https://github.com/vitejs/vite/issues/20299
+  process.env.NODE_ENV ??= 'development';
   const config = await loadConfig();
   const host = flags.host;
   const port = parseInt(flags.port || '3000', 10);

--- a/packages/waku/src/lib/vite-rsc/cmd-dev.ts
+++ b/packages/waku/src/lib/vite-rsc/cmd-dev.ts
@@ -79,7 +79,7 @@ async function startDevServer(
 }
 
 export async function runDev(flags: { host?: string; port?: string }) {
-  // before vite.runnerImport: https://github.com/vitejs/vite/issues/20299
+  // set NODE_ENV before vite.runnerImport: https://github.com/vitejs/vite/issues/20299
   process.env.NODE_ENV ??= 'development';
   const config = await loadConfig();
   const host = flags.host;

--- a/packages/waku/src/lib/vite-rsc/cmd-start.ts
+++ b/packages/waku/src/lib/vite-rsc/cmd-start.ts
@@ -1,12 +1,13 @@
 import net from 'node:net';
 import path from 'node:path';
 import { pathToFileURL } from 'node:url';
-import { loadConfig, loadDotEnv, overrideNodeEnv } from './loader.js';
+import { loadConfig, loadDotEnv } from './loader.js';
 
 loadDotEnv();
 
 export async function runStart(flags: { host?: string; port?: string }) {
-  overrideNodeEnv('production');
+  // before vite.runnerImport: https://github.com/vitejs/vite/issues/20299
+  process.env.NODE_ENV ??= 'production';
   const config = await loadConfig();
   const host = flags.host;
   const port = await getFreePort(parseInt(flags.port || '8080', 10));

--- a/packages/waku/src/lib/vite-rsc/cmd-start.ts
+++ b/packages/waku/src/lib/vite-rsc/cmd-start.ts
@@ -6,7 +6,7 @@ import { loadConfig, loadDotEnv } from './loader.js';
 loadDotEnv();
 
 export async function runStart(flags: { host?: string; port?: string }) {
-  // before vite.runnerImport: https://github.com/vitejs/vite/issues/20299
+  // set NODE_ENV before vite.runnerImport: https://github.com/vitejs/vite/issues/20299
   process.env.NODE_ENV ??= 'production';
   const config = await loadConfig();
   const host = flags.host;

--- a/packages/waku/src/lib/vite-rsc/loader.ts
+++ b/packages/waku/src/lib/vite-rsc/loader.ts
@@ -18,13 +18,3 @@ export async function loadConfig(): Promise<Required<Config>> {
   }
   return resolveConfig(config);
 }
-
-export function overrideNodeEnv(nodeEnv: 'development' | 'production') {
-  // set NODE_ENV before runnerImport https://github.com/vitejs/vite/issues/20299
-  if (process.env.NODE_ENV && process.env.NODE_ENV !== nodeEnv) {
-    console.warn(
-      `Warning: NODE_ENV is set to '${process.env.NODE_ENV}', but overriding it to '${nodeEnv}'.`,
-    );
-  }
-  process.env.NODE_ENV = nodeEnv;
-}

